sema4ai-http-helper/src/sema4ai_http/__init__.py

```diff
@@ -14,60 +16,169 @@
+        profile_config = self._get_current_config_profile()
+
+        ssl_context_factory = SSLContextFactory(profile_config)
+        ssl_context = ssl_context_factory._build_ssl_context(ssl.PROTOCOL_TLS_CLIENT)
+
+        if "proxy-settings" in profile_config:
+            proxy_settings = profile_config["proxy-settings"]
+            self.connection_pool = urllib3.ProxyManager(
+                proxy_url=proxy_settings.get("https-proxy")
+                or proxy_settings.get("http-proxy"),
+                ssl_context=ssl_context,
+            )
+        else:
+            self.connection_pool = urllib3.PoolManager(ssl_context=ssl_context)
+
+        return self.connection_pool
 
 
 @lru_cache
-def _get_default_pool():
-    import ssl
+def get_network_config() -> urllib3.PoolManager | urllib3.ProxyManager:
+    return NetworkConfig().connection_pool
+
 
-    return urllib3.PoolManager(
-        ssl_context=build_ssl_context(ssl.PROTOCOL_TLS_CLIENT),
-        cert_reqs="CERT_REQUIRED",
+def build_ssl_context(
+    protocol: int | None = None, *, enable_legacy_server_connect: bool | None = None
+) -> "ssl.SSLContext":
+    return SSLContextFactory({})._build_ssl_context(
+        protocol, enable_legacy_server_connect
     )
 
 
+class ResponseWrapper:
+    def __init__(self, response: urllib3.BaseHTTPResponse) -> None:
+        self.response = response
+
+    @property
+    def status(self) -> int:
+        return self.response.status
+
+    @property
+    def status_code(self) -> int:
+        return self.response.status
+
+    @property
+    def text(self) -> str:
+        charset = "utf-8"
+        content_type = self.response.headers.get("content-type", "")
+        if "charset=" in content_type:
+            charset = content_type.split("charset=")[-1].strip()
+        return self.response.data.decode(charset, errors="replace")
+
+    @property
+    def data(self) -> bytes:
+        return self.response.data
+
+    def read(self, amt=None, decode_content=None, cache_content=False) -> bytes:
+        return self.response.read(amt, decode_content, cache_content)
+
+    def json(self) -> dict:
+        return self.response.json()
+
+    def release_conn(self) -> None:
+        self.response.release_conn()
+
+    def getheaders(self) -> urllib3.response.HTTPHeaderDict:
+        return self.response.getheaders()
+
+    def close(self) -> None:
+        self.response.close()
+
+    def raise_for_status(self) -> None:
+        if self.response.status >= 400:
+            raise urllib3.exceptions.HTTPError(
+                f"HTTP {self.response.status}: {self.response.reason}"
+            )
+
+    def ok(self) -> bool:
+        return 200 <= self.response.status < 400
+
+
+class DownloadStatus(Enum):
+    """Status"""
+
+    DONE = auto()
+    PARTIAL = auto()
+    ALREADY_EXISTS = auto()
+    HTTP_ERROR = auto()
+
+
+connection_manager = NetworkConfig().connection_pool
```

Comment:
Make this lazy (i.e.: make it a method with `lru_cache`. Also, do we want to make that public? I'd just make it private for now -- i.e.: `_get_connection_manager`).